climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1165 @@
+
+Define WildFire class.
+"""
+
+__all__ = ['WildFire']
+
+import logging
+import warnings
+import itertools
+from datetime import date
+from functools import partial
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+from scipy import sparse
+from sklearn.neighbors import BallTree
+from sklearn.cluster import DBSCAN
+import numba
+import geopandas as gpd
+from shapely.geometry import Point
+from shapely.ops import transform
+import pyproj
+
+from climada.hazard.centroids.centr import Centroids
+from climada.hazard.base import Hazard
+from climada.hazard.tag import Tag as TagHazard
+from climada.util.constants import ONE_LAT_KM, DEF_CRS
+from climada.util.dates_times import str_to_date
+
+from climada.util.alpha_shape import alpha_shape, plot_polygon
+warnings.simplefilter(action='ignore', category=FutureWarning)
+
+LOGGER = logging.getLogger(__name__)
+
+HAZ_TYPE = 'WF'
+""" Hazard type acronym for Wild Fire, might be changed to WFseason or WFsingle """
+
+CLEAN_THRESH = 30
+""" Minimal confidence value for the data from MODIS instrument to be use as input"""
+
+RES_DATA = 1.0
+""" Resolution of the data if no data origin provided (km) """
+
+BLURR_STEPS = 4
+""" steps with exponential decay for fire propagation matrix """
+
+class WildFire(Hazard):
+
+    """Contains wild fires.
+
+    Wildfires comprise the challange that the definition of an event is unclear.
+    Reporting standards vary accross regions and over time. Hence, to have
+    consistency, we consider an event as a whole fire season. This allows
+    consistent risk assessment across the global and over time. Events that
+    refer to a fire season have the tag 'WFseason'.
+
+    In order to perform concrete case studies or calibrate impact functions,
+    events can be displayed as single fires. In that case they have the tag
+    'WFsingle'.
+
+
+    Attributes:
+        date_end ((np.array): integer date corresponding to the proleptic
+            Gregorian ordinal, where January 1 of year 1 has ordinal 1
+            (ordinal format of datetime library))
+
+        n_fires (np.array): number of single fires in a fire season
+
+    """
+
+    days_thres_firms = 2
+    """ Minimum number of days to consider different fires """
+
+    clus_thres_firms = 15
+    """ Clustering factor which multiplies instrument resolution """
+
+    remove_minor_fires_firms = True
+    """ removes FIRMS fires below defined theshold of entries"""
+
+    minor_fire_thres_firms = 3
+    """ number of FIRMS entries required to be considered a fire """
+
+    prop_proba = 0.21
+    """ probability of fire propagation """
+
+    max_it_propa = 500000
+    """ maximum propgation iterations for a probabilistic fire """
+
+    def __init__(self, pool=None):
+        """Empty constructor. """
+        Hazard.__init__(self, HAZ_TYPE)
+        if pool:
+            self.pool = pool
+            LOGGER.info('Using %s CPUs.', self.pool.ncpus)
+        else:
+            self.pool = None
```

Comment:
Why would a pool of workers be initiated when the object is initiated? There is no computation to be done here, so I do not see a reason to set a pool of workers. 
```suggestion
    def __init__(self):
        """Empty constructor. """
        Hazard.__init__(self, HAZ_TYPE)
```

Reply:
This part was implemented by Gabriela (for a paralysation strategy). Happy to remove it if you don't see use for it

Reply:
Yes remove it. However, if somewhere a pool is used for computation, do not forget to add the pool there!

Reply:
it's removed